### PR TITLE
Sensorintegration

### PR DESCRIPTION
**Pull request by AnotherOctopus**
Integrates sensors

### CODE REVIEW

**Review by bmaxfie — CHANGES_REQUESTED** [review]
Along with comments in code. What are the recommended update frequencies for the IMU and Pressure sensor

File: rov/rov.py [hunk 1]
```diff
@@ -62,10 +62,8 @@ def init_hw(self):
 
         self.controls = controller(self.motor_control)
 
-        #""" Disabled until hardware is done and sw is tested
-        # self.IMU = IMU()
-        # self.pressure = Pressure()
-        #"""
+        self.IMU = IMU()
```
Comment (bmaxfie) [inline rov/rov.py]: Change self.IMU to self.imu

To keep to the convention of lowercase instances.

File: rov/rov.py [hunk 2]
```diff
@@ -77,23 +75,19 @@ def update(self):
             # self.thruster_control.stop()
 
         try:
+	    self.IMU.update()
```
Comment (bmaxfie) [inline rov/rov.py]: Above

File: rov/rov.py [hunk 3]
```diff
@@ -77,23 +75,19 @@ def update(self):
...
         except Exception as e:
             print "Failed updating things"
             print "Exception: %s" % e
             print traceback.format_exc()
 
 
-        # retrieve all sensor data
-        # self.dearclient['sensor'] = sensorThings
-
+        self.dearclient['imu'] = self.IMU.data
```
Comment (bmaxfie) [inline rov/rov.py]: Above

File: rov/sensors/pressure/BlueRobotics_Bar30.py [hunk 4]
```diff
@@ -1,100 +1,42 @@
 from smbus import SMBus
+import ms5837
 import time
-
 class Pressure(object):
-
-    def __init__(self):
+    
+    def __init__(self,PressDepthConv=0.01):
```
Comment (bmaxfie) [inline rov/sensors/pressure/BlueRobotics_Bar30.py]: If PressDepthConv is useful to us, add comments saying how to use it!

File: rov/sensors/pressure/BlueRobotics_Bar30.py [hunk 5]
```diff
@@ -1,100 +1,42 @@
...
-
-        # Read temperature coefficient of the temperature
-        data = self.bus.read_i2c_block_data(0x76, 0xAC, 2)
-        C6 = data[0] * 256 + data[1]
-
-        # MS5837_30BA01 address, 0x76(118)
-        #0x40(64) Pressure conversion(OSR = 256) command
-        self.bus.write_byte(0x76, 0x40)
-
         time.sleep(0.5)
```
Comment (bmaxfie) [inline rov/sensors/pressure/BlueRobotics_Bar30.py]: You left this line here??? Why?

File: rov/sensors/pressure/README.md [hunk 6]
```diff
@@ -1,13 +1,120 @@
-# Pressure Sensor
+# ms5837-python
 
-This contains the pressure sensor library for the BlueRobotics Bar30 Pressure Sensor. The Bar30
-uses a MS5837-30BA01 Pressure and Temperature Sensor.
- - [Product Details](https://www.bluerobotics.com/store/electronics/bar30-sensor-r1/)
- - [Reference Code](https://github.com/ControlEverythingCommunity/MS5837-30BA01)
+A python module to interface with MS5837-30BA and MS5837-02BA waterproof pressure and temperature sensors. Tested on Raspberry Pi 3 with Raspbian.
 
-### Return Data:
```
Comment (bmaxfie) [inline rov/sensors/pressure/README.md:8]: A++++++ README THO

File: surface/frontend/src/packets.js [hunk 7]
```diff
@@ -46,19 +63,6 @@ module.exports = {
             roll: false,
             yaw: false
         },
-        pressure: {
```
Comment (bmaxfie) [inline surface/frontend/src/packets.js]: Where does pressure go now? Or at least what is its format?

File: tests/rov_pressure_test.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+#print("Temperature: %.2f C") % (sensor.temperature(ms5837.UNITS_Centigrade))
+
+#time.sleep(2)
+
+print("Time \tPressure (mbar) \tTemperature (C)")
+
+# Spew readings
+while True:
+        if sensor.read():
+            print("%s \t%0.1f \t%0.2f") % (time.strftime("%H:%M:%S", time.localtime()) + '.%d' % (time.time() % 1 * 1000),
```
Comment (bmaxfie) [inline tests/rov_pressure_test.py]: So the tests/ folder is primarily for Unit Tests. This seems more like just a start script. This is good to have. I would put this exact code at the bottom of the pressure code under a `if __name__ == '__main__':` so we can make it run by running the ms5837 file. And make another for the tests/ folder that just checks whether a reasonable value is generated from the sensor and whether it changes. I think that would be a good enough Unit Test that the sensor code is working (for automated testing).

File: rov/sensors/pressure/BlueRobotics_Bar30.py [hunk 9]
```diff
@@ -1,100 +1,42 @@
 from smbus import SMBus
+import ms5837
 import time
-
 class Pressure(object):
```
Comment (bmaxfie) [inline rov/sensors/pressure/BlueRobotics_Bar30.py]: What is the difference between this file and the ms5837 file?

**Review by bmaxfie — CHANGES_REQUESTED** [review]
These are the changes I was talking about above... Whoops!

File: rov/sensors/imu/BNO055.py [hunk 10]
```diff
@@ -44,7 +46,7 @@ def update(self):
         self._data['euler']['roll']    = euler[1]
         self._data['euler']['pitch']   = euler[2]
 
-        gyro = self._bno.read_gyroscrope()
+	gyro = self._bno.read_gyroscope()
```
Comment (mattmolo) [inline rov/sensors/imu/BNO055.py:49]: there's a tab here

**Comment by bmaxfie** [comment]
@MoldingTofu Confirmation the sensor works? I know the frequency/accuracy test hasn't been performed yet, but I can merge as long as its working.

**Comment by AnotherOctopus** [comment]
@bmaxfie I did the frequency test, with an accuracy check using the pressure chambers gauge on top, and it seemed to be fine. I posted about it in SW general

**Comment by bmaxfie** [comment]
I see it now. This is good. Okay. Refresh rate of 5 ms is perfect.

**Comment by bmaxfie** [comment]
Just need IMU to imu fix and I think it'll be set?

**Comment by AnotherOctopus** [comment]
@bmaxfie IMU update pushed

**Review by bmaxfie — APPROVED** [review]

**Pull request merged**